R/calc_truth.R

```diff
@@ -0,0 +1,328 @@
+#' Calculate "truth" for a stock using {ecosystemdata} outputs
+#' 
+#' @description This function generates standardized "truth" outputs using data 
+#' from {ecosystemdata}. The "truth" include "true" index and age composition
+#' (both monthly and yearly) for a specified stock.
+#' 
+#' @details Currently, this function supports calculating "truth" for biomass, 
+#' catch, numbers, and weight. Numbers-at-age are calculated by dividing 
+#' biomass-at-age by weight-at-age.
+#' 
+#' @param data A tibble containing ecosystem model (e.g., Ecopath with Ecosim) 
+#' outputs from {ecosystemdata}.
+#' @param stock_name A character string specifying the stock name to subset.
+#' 
+#' @return A tibble with nested "truth" (e.g., biomass, catch, 
+#' numbers, or weight), categorized by type (index or age composition) and 
+#' timestep (monthly or yearly).
+#' 
+#' @examples
+#' data(ewe_nwatlantic_base, package = "ecosystemdata")
+#' data <- ewe_nwatlantic_base
+#' truth <- calc_truth(
+#'   data,
+#'   stock_name = "menhaden"
+#' )
+#' @export
+#' @rdname calc_truth
+#' @keywords calc_truth
+calc_truth <- function(data, stock_name){
+
+  # Check if the stock_name is in the data
+  if (!stock_name %in% unique(data[["species"]])) {
+    cli::cli_abort(c(
+      "Stock {.val {stock_name}} not found in the data.",
+      "Please check the stock name and try again."
```

Comment:
```suggestion
      "x" = "Stock {.val {stock_name}} not found in the data.",
      "i" = "Please check the stock name and try again."
```
You could list the available ones or ones that start with the same letter in the "information" that is returned on line 35.

Reply:
Done. I have listed the available ones.